src/SignalR/clients/csharp/Client.SourceGenerator/src/HubProxyGenerator.Emitter.cs

```diff
@@ -0,0 +1,178 @@
+                {
+                    var signature = new StringBuilder($"public {methodSpec.FullyQualifiedReturnTypeName} {methodSpec.Name}(");
+                    var callArgs = new StringBuilder("");
+                    var signatureArgs = new StringBuilder("");
+                    var first = true;
+                    foreach (var argumentSpec in methodSpec.Arguments)
+                    {
+                        if (!first)
+                        {
+                            signatureArgs.Append(", ");
+                        }
+
+                        first = false;
+                        signatureArgs.Append($"{argumentSpec.FullyQualifiedTypeName} {argumentSpec.Name}");
+                        callArgs.Append($", {argumentSpec.Name}");
+                    }
+                    signature.Append(signatureArgs);
+                    signature.Append(')');
+
+                    // Prepare method body
+                    var body = "";
+                    if (methodSpec.Support != SupportClassification.Supported)
+                    {
+                        body = methodSpec.SupportHint is null
+                            ? "throw new System.NotSupportedException();"
+                            : $"throw new System.NotSupportedException(\"{methodSpec.SupportHint}\");";
+                    }
+                    else
+                    {
+                        // Get specific hub connection extension method call
+                        var specificCall = getSpecificCall(methodSpec);
+
+                        // Handle ValueTask
+                        var prefix = "";
+                        var suffix = "";
+                        if (methodSpec.IsReturnTypeValueTask)
+                        {
+                            if (methodSpec.InnerReturnTypeName is not null)
+                            {
+                                prefix = $"new System.Threading.Tasks.ValueTask<{methodSpec.InnerReturnTypeName}>(";
+                            }
+                            else
+                            {
+                                prefix = "new System.Threading.Tasks.ValueTask(";
+                            }
+                            suffix = $")";
+                        }
+
+                        // Bake it all together
+                        body = $"return {prefix}this.connection.{specificCall}(\"{methodSpec.Name}\"{callArgs}){suffix};";
+                    }
+
+                    var method = $@"
+        {signature}
+        {{
+            {body}
+        }}
+";
+                    methods.Append(method);
+                }
+
+                var proxy = $@"// Generated by Microsoft.AspNetCore.Client.SourceGenerator
+using Microsoft.AspNetCore.SignalR.Client;
+namespace Microsoft.AspNetCore.SignalR.Client.SourceGenerated
+{{
+    public sealed class {classSpec.ClassTypeName} : {classSpec.FullyQualifiedInterfaceTypeName}
```

Comment:
@BrennanConroy Do we care if the generated proxy is public?

On my original repo, someone ran into compiler errors with `/doc` compiler flag enabled as these generated classes were missing documentation. I was going to add docstrings to `GetProxy` `RegisterCallbackProvider` but I wasn't planning to for the proxies initially.